src/app/WriteHandler.cpp

```diff
@@ -829,6 +829,16 @@ DataModel::ActionReturnStatus WriteHandler::CheckWriteAllowed(const Access::Subj
     VerifyOrReturnValue(IsTimedWrite() || !attributeEntry->HasFlags(DataModel::AttributeQualityFlags::kTimed),
                         Status::NeedsTimedInteraction);
 
+    if (aPath.mDataVersion.HasValue())
```

Comment:
Followup: Between the timed write check an this check, the spec has the "write to a fabric-scoped list" check.  This check is missing here (and I don't think it's done anywhere....).